scylla-server-rust/src/controller/data_controller.rs

```diff
@@ -0,0 +1,42 @@
+use axum::{
+    extract::{Path, State},
+    Json,
+};
+use serde::Serialize;
+
+use crate::{
+    error::ScyllaError,
+    prisma::{self},
+    services::data_service,
+    Database,
+};
+
+#[derive(Serialize, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct DataSend {
+    pub time: i64,
+    pub values: Vec<String>,
+}
+
+// convert the prisma type to the client type for JSON encoding
+impl From<&prisma::data::Data> for DataSend {
```

Comment:
I would move this struct and implementation to a transformers folder, same comments for all the public types